### PR TITLE
Bump cloud provider to 1.13 and fix CSI Driver kube-version

### PR DESCRIPTION
**Pull request by chrisho**
Bump cloud provider to 1.13 and fix CSI Driver `kube-version: >= 1.18.0-0 < 1.25.0-0`

related issue: https://github.com/rancher/rancher/issues/38618

### CODE REVIEW

**Review by bk201 — DISMISSED** [review]
approve for running workflow

**Comment by rosskirkpat** [comment]
approved to run the workflow and tagged some of the @rancher/rancher2-helm-yes-dev members

**Comment by rosskirkpat** [comment]
@chrisho Please take a look at the failing workflow.

```log
time="2022-08-15T17:10:47Z" level=info msg="Generated index.yaml"
time="2022-08-15T17:10:47Z" level=info msg="Comparing standardized upstream assets against local assets"
time="2022-08-15T17:10:47Z" level=info msg="harvester-cloud-provider/100.0.3+up0.1.13 is untracked"
time="2022-08-15T17:10:50Z" level=error msg="The following new assets have been introduced: map[harvester-cloud-provider:[100.0.3+up0.1.13]]"
time="2022-08-15T17:10:50Z" level=error msg="The following released assets have been removed: map[]"
time="2022-08-15T17:10:50Z" level=error msg="The following released assets have been modified: map[]"
time="2022-08-15T17:10:50Z" level=error msg="If this was intentional, to allow validation to pass, these charts must be added to the release.yaml."
time="2022-08-15T17:10:50Z" level=info msg="Dumping release.yaml tracking changes that have been introduced"
time="2022-08-15T17:10:50Z" level=info msg="Updating index.yaml"
time="2022-08-15T17:10:50Z" level=info msg="index.yaml is up-to-date"
time="2022-08-15T17:10:50Z" level=fatal msg="Validation against upstream repository https://github.com/rancher/charts.git at branch release-v2.6 failed."
```

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Comment by aiyengar2** [comment]
@chrisho this should pass after you add a fix into https://github.com/rancher/charts/blob/e3b103441cb6bfd27d76e1e5381ee44b3e52b54a/release.yaml#L10-L11 to add the CPI chart to it

File: charts/harvester-csi-driver/100.0.3+up0.1.14/Chart.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 annotations:
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Harvester CSI Driver
-  catalog.cattle.io/kube-version: '>= 1.18.0-0 < 1.24.0-0'
+  catalog.cattle.io/kube-version: '>= 1.18.0-0 < 1.25.0-0'
   catalog.cattle.io/namespace: kube-system
   catalog.cattle.io/os: linux
-  catalog.cattle.io/permits-os: linux
+  catalog.cattle.io/permits-os: linux,windows
```
Comment (aiyengar2) [inline charts/harvester-csi-driver/100.0.3+up0.1.14/Chart.yaml:7]: missing upstream annotation? https://github.com/rancher/charts/blob/dev-v2.6/charts/harvester-csi-driver/100.0.2%2Bup0.1.11/Chart.yaml#L11

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

**Comment by aiyengar2** [comment]
The upstream versions in the following two released charts are also incorrect, should probably be fixed with this release:

https://github.com/rancher/charts/blob/540def26e58b341c6f79f5d1a93a4379d9a03b8f/charts/harvester-cloud-provider/100.0.2%2Bup0.1.12/Chart.yaml#L11

https://github.com/rancher/charts/blob/540def26e58b341c6f79f5d1a93a4379d9a03b8f/charts/harvester-csi-driver/100.0.2%2Bup0.1.11/Chart.yaml#L11

**Comment by yasker** [comment]
Approved for CI rerun.

**Review by aiyengar2 — APPROVED** [review]
```diff
diff -uNr charts/harvester-cloud-provider/100.0.2+up0.1.12/Chart.yaml charts/harvester-cloud-provider/100.0.3+up0.1.13/Chart.yaml
--- charts/harvester-cloud-provider/100.0.2+up0.1.12/Chart.yaml 2022-08-16 09:43:24.000000000 -0700
+++ charts/harvester-cloud-provider/100.0.3+up0.1.13/Chart.yaml 2022-08-16 09:43:24.000000000 -0700
@@ -8,9 +8,9 @@
   catalog.cattle.io/rancher-version: '>= 2.6.1-0 < 2.7.0-0'
   catalog.cattle.io/release-name: harvester-cloud-provider
   catalog.cattle.io/ui-component: harvester-cloud-provider
-  catalog.cattle.io/upstream-version: 0.1.12
+  catalog.cattle.io/upstream-version: 0.1.13
 apiVersion: v2
-appVersion: v0.1.3
+appVersion: v0.1.4
 description: A Helm chart for Harvester Cloud Provider
 keywords:
 - infrastructure
@@ -19,4 +19,4 @@
 - name: harvester
 name: harvester-cloud-provider
 type: application
-version: 100.0.2+up0.1.12
+version: 100.0.3+up0.1.13
diff -uNr charts/harvester-cloud-provider/100.0.2+up0.1.12/values.yaml charts/harvester-cloud-provider/100.0.3+up0.1.13/values.yaml
--- charts/harvester-cloud-provider/100.0.2+up0.1.12/values.yaml        2022-08-15 11:40:15.000000000 -0700
+++ charts/harvester-cloud-provider/100.0.3+up0.1.13/values.yaml        2022-08-16 09:43:24.000000000 -0700
@@ -8,7 +8,7 @@
   repository: rancher/harvester-cloud-provider
   pullPolicy: IfNotPresent
   # Overrides the image tag whose default is the chart appVersion.
-  tag: v0.1.3
+  tag: v0.1.4
 
 cloudConfigPath: "/etc/kubernetes/cloud-config"
```


```diff
diff -uNr charts/harvester-csi-driver/100.0.2+up0.1.11/Chart.yaml charts/harvester-csi-driver/100.0.3+up0.1.14/Chart.yaml
--- charts/harvester-csi-driver/100.0.2+up0.1.11/Chart.yaml     2022-08-16 09:43:24.000000000 -0700
+++ charts/harvester-csi-driver/100.0.3+up0.1.14/Chart.yaml     2022-08-16 09:43:24.000000000 -0700
@@ -8,7 +8,7 @@
   catalog.cattle.io/rancher-version: '>= 2.6.1-0 < 2.7.0-0'
   catalog.cattle.io/release-name: harvester-csi-driver
   catalog.cattle.io/ui-component: harvester-csi-driver
-  catalog.cattle.io/upstream-version: 0.1.11
+  catalog.cattle.io/upstream-version: 0.1.14
 apiVersion: v2
 appVersion: v0.1.3
 description: A Helm chart for Harvester CSI driver
@@ -19,4 +19,4 @@
 - name: harvester
 name: harvester-csi-driver
 type: application
-version: 100.0.2+up0.1.11
+version: 100.0.3+up0.1.14
diff -uNr charts/harvester-csi-driver/100.0.2+up0.1.11/values.yaml charts/harvester-csi-driver/100.0.3+up0.1.14/values.yaml
--- charts/harvester-csi-driver/100.0.2+up0.1.11/values.yaml    2022-08-15 11:40:15.000000000 -0700
+++ charts/harvester-csi-driver/100.0.3+up0.1.14/values.yaml    2022-08-15 12:58:57.000000000 -0700
@@ -7,7 +7,7 @@
     csiDriver:
       repository: rancher/harvester-csi-driver
       # Overrides the image tag whose default is the chart appVersion.
-      tag: ""
+      tag: "v0.1.3"
   csi:
     nodeDriverRegistrar:
       repository: rancher/mirrored-longhornio-csi-node-driver-registrar
```

Diff across versions looks good to me and OOB changes to existing charts LGTM.

**Pull request merged**